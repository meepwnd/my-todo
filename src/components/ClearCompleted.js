import React from "react";
import { connect } from "react-redux";
import { clearCompleted } from "../actions/actions";

const ClearCompletedButton = ({ clearCompleted }) => (
  <button className="btn btn-clear-completed" onClick={clearCompleted}>
    Clear completed
  </button>
);

export default connect(
  null,
  { clearCompleted }
)(ClearCompletedButton);
