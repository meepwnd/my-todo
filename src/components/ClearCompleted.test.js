import React from 'react';
import {shallow} from 'enzyme'
import {ClearCompletedButton} from './ClearCompleted'

describe('clear completed button', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ClearCompletedButton />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call clear completed function after button click', () => {
    const clearCompletedSpy = jest.fn()
    const wrapper = shallow(<ClearCompletedButton clearCompleted={clearCompletedSpy}/>)
    wrapper.find('button').simulate('click')
    expect(clearCompletedSpy).toHaveBeenCalled()
  })
})