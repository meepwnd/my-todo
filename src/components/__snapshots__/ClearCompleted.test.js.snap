// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clear completed button should render correctly 1`] = `
<button
  className="btn btn-clear-completed"
>
  Clear completed
</button>
`;
